### PR TITLE
Fix host image builds in Caracal

### PR DESCRIPTION
**Pull request by Alex-Welsh**

### CODE REVIEW

**Comment by markgoddard** [comment]
Quite a lot going on here

**Comment by Alex-Welsh** [comment]
> Quite a lot going on here

The only "real" changes are the 4 latest commits.

I based my changes on your branch, but also synced the latest stackhpc/2024.1 in.

File: etc/kayobe/stackhpc-overcloud-dib.yml [hunk 1]
```diff
@@ -47,6 +47,8 @@ stackhpc_overcloud_dib_env_vars:
   # sometimes).
   # DIB_DISTRIBUTION_MIRROR: "{{ stackhpc_repo_ubuntu_focal_url if os_distribution == 'ubuntu' else '' }}"
   DIB_DRACUT_ENABLED_MODULES_DEFAULT_CONFIG: "{{ stackhpc_overcloud_dib_dracut_enabled_modules_default_config }}"
+  # NOTE: DIB_ETC_HOSTS_EXTRA must be defined and cannot be an empty string 
+  DIB_ETC_HOSTS_EXTRA: "#"
```
Comment (markgoddard) [inline etc/kayobe/stackhpc-overcloud-dib.yml:51]: Better to remove L28?

**Comment by Alex-Welsh** [comment]
This branch has become a mess. I'll close this PR and cherry-pick out the good bits.

**Pull request closed without merging**